web/files/files.go

```diff
@@ -714,13 +716,72 @@ func DestroyFileHandler(c echo.Context) error {
 	return c.NoContent(204)
 }
 
+const maxMangoLimit = 100
+
+// FindFilesMango is the route POST /files/_find
+// used to retrieve files and their metadata from a mango query.
+func FindFilesMango(c echo.Context) error {
+	instance := middlewares.GetInstance(c)
+	var findRequest map[string]interface{}
+
+	if err := c.Bind(&findRequest); err != nil {
+		return jsonapi.NewError(http.StatusBadRequest, err)
+	}
+
+	if err := permissions.AllowWholeType(c, permissions.GET, consts.Files); err != nil {
+		return err
+	}
+
+	limit, hasLimit := findRequest["limit"].(float64)
+	if !hasLimit || limit > maxMangoLimit {
+		limit = 100
+	}
+	skip := 0
+	skipF64, hasSkip := findRequest["skip"].(float64)
+	if hasSkip {
+		skip = int(skipF64)
+	}
+
+	// add 1 so we know if there is more.
+	findRequest["limit"] = limit + 1
+
+	var results []vfs.DirOrFileDoc
+	err := couchdb.FindDocsRaw(instance, consts.Files, &findRequest, &results)
+	if err != nil {
+		return err
+	}
+
+	var total int
+	if len(results) > int(limit) {
+		total = math.MaxInt32 - 1          // we dont know the actual number
```

Comment:
Agree this is not ideal:
- We dont and cant know the total number of results from a mango query.
- We want to let the client-side know when there is no more results, with JSON-APi convention
- This solutions returns a number which is higher than any possible number, so we cant simply do client-side `if(total > received)`